compiler/src/iree/compiler/Dialect/Flow/Transforms/test/deduplicate_executables.mlir

```diff
@@ -375,3 +375,55 @@ flow.executable @block_successors_ex_with_swapped_cond_br {
     }
   }
 }
+
+// -----
+
+// CHECK: hal.executable private @ex0
+hal.executable private @ex0 {
```

Comment:
Feels weird to test `hal` ops in `Dialect/Flow`

> Future changes may move the pass out of flow and into util so we can run it at multiple points during compilation.

Seems like a good idea in general.

Reply:
yeah, the HAL external dispatch op is a frontend thing expected to be in inputs from global optimization/plugins and we will see it here today. We could create a new dialect and move it there to decouple things but it relies on HAL attributes so we'd still be indirectly pulling in HAL. Moving this pass to util would let us make mixed hal/flow an integration test instead of a pass test, at least, but testing the pass without the ops is really annoying as we'd need test ops in util in order to have the test there but not depend on flow/hal. The next PR will add outlining of the external dispatch ops into hal.executable ops (why this PR was needed) and that also bothers me but there's currently no good place to put that either without joining flow/hal (as we want to insert flow.dispatch but create hal.executables). Definitely on the cleanup list and mostly wanted to get something working end-to-end before trying to figure out the right location for things.

Reply:
SGTM. Is "the cleanup list" in an issue somewhere?

Reply:
![image](https://github.com/openxla/iree/assets/75337/bbabc51d-8049-4c34-b850-b7202310d5c6)

I think I'm waiting for Hanhan's cleanups around flow/global optimization to finish before I do it - but if you're interested in helping out I can file something >:)